contracts/StrategyCurveTwoPool.sol

```diff
@@ -0,0 +1,372 @@
+    uint256 _stakedBal = stakedBalance();
+    if (_stakedBal > 0) {
+      gauge.withdraw(_stakedBal);
+    }
+  }
+
+  function protectedTokens()
+    internal
+    view
+    override
+    returns (address[] memory)
+  {}
+
+  /* ========== SETTERS ========== */
+
+  // These functions are useful for setting parameters of the strategy that may need to be adjusted.
+
+  // Credit threshold is in want token, and will trigger a harvest if credit is large enough.
+  function setCreditThreshold(uint256 _creditThreshold)
+    external
+    onlyEmergencyAuthorized
+  {
+    creditThreshold = _creditThreshold;
+  }
+
+  // This allows us to manually harvest with our keeper as needed
+  function setForceHarvestTriggerOnce(bool _forceHarvestTriggerOnce)
+    external
+    onlyEmergencyAuthorized
+  {
+    forceHarvestTriggerOnce = _forceHarvestTriggerOnce;
+  }
+
+  /* ========== KEEP3RS ========== */
+
+  /// @param callCostInWei Cost of the contract call to harvest(). Used to determine if the profit (if any) is
+  ///   sufficiently large to justify the call to harvest()
+  function harvestTrigger(uint256 callCostInWei)
+    public
+    view
+    override
+    returns (bool)
+  {
+    // Should not trigger if strategy is not active (no assets and no debtRatio). This means we don't need to adjust keeper job.
+    if (!isActive()) {
+      return false;
+    }
+
+    StrategyParams memory params = vault.strategies(address(this));
+
+    // harvest no matter what once we reach our maxDelay
+    if (block.timestamp.sub(params.lastReport) > maxReportDelay) {
+      return true;
+    }
+
+    // harvest our credit if it's above our threshold
+    if (vault.creditAvailable() > creditThreshold) {
+      return true;
+    }
+
+    // trigger if we want to manually harvest
+    if (forceHarvestTriggerOnce) {
+      return true;
+    }
+
+    // otherwise, we don't harvest
+    return false;
+  }
+
+  // convert our keeper's eth cost into want, we don't need this anymore since we don't use baseStrategy harvestTrigger
+  function ethToWant(uint256 _ethAmount)
+    public
+    view
+    override
+    returns (uint256)
+  {
+    return _ethAmount;
+  }
+}
+
+contract StrategyCurveTwoPool is StrategyCurveBase {
+  /* ========== STATE VARIABLES ========== */
+  // `targetTokenAddress` is the token we buy with CRV rewards and the token we then
+  // deposit into the curve pool -- USDT or USDC
+  address public targetTokenAddress;
+
+  IERC20 internal usdt;
+  IERC20 internal usdc;
+
+  uint24 public crvToWethSwapFee;
+  uint24 public wethToTargetSwapFee;
+  uint24 internal constant maxFee = 10000;
+
+  /* ============= CONSTRUCTOR =========== */
+  constructor(
+    address _vault,
+    string memory _name,
+    address _usdtAddress,
+    address _usdcAddress,
+    address _healthCheckAddress,
+    address _gaugeAddress,
+    address _poolAddress,
+    address _wethAddress,
+    address _crvAddress,
+    address _routerAddress
+  )
+    public
+    StrategyCurveBase(
```

Comment:
weird constructor, some variables are set here, some are set in the body, for example, why `stratName` is not passed here instead of line 251?

Reply:
I agree. I tried to split the code into state variables and functions that could potentially apply to all curve strategies and then variables and functions that only apply to curve's two pool. `stratName` is the only variable that belonged to the strategic-specific group but that I did not move because it was needed by name() in the abstract contract (it would not compile). To be 100% consistent, though, I've now moved this function to the main contract along with `stratName`.  I did not identify other variables with a similar issue.